src/imfill.jl

```diff
@@ -0,0 +1,39 @@
+"""
+ Image filling Algorithm
+
+ ```
+ filled_img = imfill(img, interval)
+ filled_img = imfill(img, interval, value)
+ filled_img = imfill(img, interval, value, connectivity)
+ ```
+
+ Connected components of an image is found using flood-fill algorithm and returns a copy of
+ the original image after filling objects that falls in the range of interval.
+
+
+ Parameters:
+
+  -  img            = Input image (Boolean array type)
+  -  interval       = objects of size in this range will be filled with False
+  -  connectivity   = connectivity takes the same values as in label_components (Default value is 1:ndims(img))
+
+ """
+
+  function imfill(img::BitArray, interval::Tuple{Int,Int}=(0,64), connectivity::Union{Dims, AbstractVector{Int}, BitArray}=1:ndims(img))
+
+     labels = label_components(img,connectivity)
+
+     number_labels = unique(labels)
+     count_labels = Dict([(i,count(x->x==i,labels)) for i in number_labels])
+
+     new_img = similar(img)
+     for ind in eachindex(img)
+         if img[ind] == true && count_labels[labels[ind]] in interval[1]:interval[2]
```

Comment:
I thought holes were represented with `0` instead of `1`? What is the most common use case? I always treat `0` or `false` as deactivated pixels in my applications.

Also, could you please double check that the efficiency of checking `count in interval[1]:interval[2]` is the same as `interval[1] <= count <= interval[2]`?

Reply:
If you use `interval[1] <= count <= interval[2]` you could accept `interval::Tuple{Real,Real}`.

Reply:
@juliohm I agree with holes being represented with `0` but the function `label_components` in its implementation make connected components with pixels `true` or `1`. And, as we want to remove objects, the objects(holes) have to be that way. The rest of the implementation in `infill` is according to that logic.
As you suggested to reduce pre-processing in `imfill` , I think the user could handle inverting the image for removing holes.

Reply:
Thank you @aquatiko , could you please document this convention in the docstring? That is, holes are blobs with 1s as opposed to 0s? Please ignore my comment if that is already documented.

Reply:
@juliohm Would this be good enough?
```
Connected components of an image is found using flood-fill algorithm and returns a copy of
the original image after filling objects that falls in the range of interval.
For filling holes, use True to represent holes in your array.
```

Reply:
Awesome @aquatiko :+1: , I think it is quite clear now.